docs/sources/flow/reference/components/loki.source.file.md

```diff
@@ -26,19 +26,44 @@ log entries to the list of receivers passed in `forward_to`.
 
 `loki.source.file` supports the following arguments:
 
-Name         | Type                   | Description          | Default | Required
------------- | ---------------------- | -------------------- | ------- | --------
-`targets`    | `list(map(string))`    | List of files to read from. | | yes
-`forward_to` | `list(LogsReceiver)`   | List of receivers to send log entries to. | | yes
-`encoding`   | `string`               | The encoding to convert from when reading files. | `""` | no
+ Name         | Type                 | Description                                      | Default | Required 
+--------------|----------------------|--------------------------------------------------|---------|----------
+ `targets`    | `list(map(string))`  | List of files to read from.                      |         | yes      
+ `forward_to` | `list(LogsReceiver)` | List of receivers to send log entries to.        |         | yes      
+ `encoding`   | `string`             | The encoding to convert from when reading files. | `""`    | no       
+
+The `encoding` argument must be a valid [IANA encoding][] name. If not set, it
+defaults to UTF-8.
 
 ## Blocks
 
-The `loki.source.file` component doesn't support any inner blocks and is
-configured fully through arguments.
+The following blocks are supported inside the definition of `loki.source.file`:
 
-The `encoding` argument must be a valid [IANA encoding][] name. If not set, it
-defaults to UTF-8. 
+ Hierarchy      | Name               | Description                                   | Required 
+----------------|--------------------|-----------------------------------------------|----------
+ decompresssion | [decompresssion][] | Configure reading logs from compressed files. | no       
+
+[decompresssion]: #decompresssion-block
+
+### decompresssion block
+
+The `decompression` block contains configuration for reading logs from 
+compressed files. The following arguments are supported:
+
+ Name            | Type       | Description                                                     | Default | Required 
+-----------------|------------|-----------------------------------------------------------------|---------|----------
+ `enabled`       | `bool`     | Whether decompression is enabled.                               |         | yes      
+ `initial_delay` | `duration` | Time to wait before starting to read from new compressed files. | 0       | no       
+ `format`        | `string`   | Compression format.                                             |         | yes      
+
+If you compress a file under a folder being scraped, `loki.source.file` might
+try to ingest your file before you finish compressing it. To avoid it, pick
+an `initial_delay` that is enough to avoid it.
```

Comment:
Since we cannot specify multiple formats, should we explicitly mention that to handle different compressions, users should add a new component?